rest/src/main/java/fi/otavanopisto/pyramus/rest/StudentRESTService.java

```diff
@@ -1816,9 +1851,23 @@ public Response updateStudentContactLogEntry(@PathParam("STUDENTID") Long studen
     if (!contactLogEntry.getStudent().getId().equals(contactLogEntry.getStudent().getId())) {
       return Response.status(Status.NOT_FOUND).build();
     }
+    
+    User loggedUser = sessionController.getUser();
+    StaffMember staffMember = userController.findStaffMemberById(loggedUser.getId());
+    Boolean amIGuidanceCounselor = studentController.amIGuidanceCouncelor(studentId, staffMember);
```

Comment:
Plenty of copy-pasting this code. Maybe have a method in `studentController` that is something like...

`public ContactLogAccess resolveContactLogAccess(Student student)`

...where the return value is an enum with values `NONE`, `OWN`, `ALL`. Administrators and student's own guidance councelors get `ALL`, other staff members (except those in role `CLOSED`) get `OWN`, and everyone else (students or closed accounts) get `NONE`.